numpyro/primitives.py

```diff
@@ -649,6 +650,105 @@ def plate_stack(
         yield
 
 
+class label_event_dim(plate):
+    """This labels event dimensions, modeled after numpyro.plate. Unlike numpyro.plate, it will not
+    change the shape of any sites within its context.
+
+    Labelled event dims can be found in `dep_stack` in the model trace.
+
+    **example**
+
+        with label_event_dim("groups", n_groups):
+            alpha = numpyro.sample("alpha", dist.ZeroSumNormal(1, event_shape=(n_groups,)))
+    """
+
+    def __init__(
+        self,
+        name: str,
+        size: int,
+        subsample_size: Optional[int] = None,
```

Comment:
still considering if size and subsample_size are needed. right now they dont do anything, but I could potentially use `size` for some form of shape validation